security/advancedtls/advancedtls.go

```diff
@@ -171,7 +171,7 @@ type ClientOptions struct {
 	// check.
 	// If this is set, we will perform this customized check after doing the
 	// normal check(s) indicated by setting VType.
-	VerifyPeer CustomVerificationFunc
+	VerifyPeer func(tls.ConnectionState) error
```

Comment:
This will be an API change. Though these APIs are in experimental and theoretically can be changed at will, I would prefer add one instead of changing it.
Also note that this function is added in Go 1.5, but gRPC Go supports version prior to 1.5. So please make sure a proper build constraint is added for versions prior and after 1.5.

Reply:
Sorry, this is actually added in Go 1.15, not 1.5. gRPC doesn't support anything older than 1.11 right now. You can find it at https://golang.org/pkg/crypto/tls/#Config.